core/server/worker/src/main/java/alluxio/worker/block/management/DefaultStoreLoadTracker.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.management;
+
+import alluxio.collections.ConcurrentHashSet;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.util.ThreadFactoryUtils;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.io.BlockReader;
+import alluxio.worker.block.io.BlockStreamListener;
+import alluxio.worker.block.io.BlockStreamTracker;
+import alluxio.worker.block.io.BlockWriter;
+
+import com.google.common.base.Preconditions;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Default {@link StoreLoadTracker} that reports load based on open reader/writer streams to the
+ * local block store.
+ *
+ * TODO(ggezer): TV2 - Add a safety net against close calls not being called.
+ */
+public class DefaultStoreLoadTracker implements StoreLoadTracker, BlockStreamListener {
+  /** Used to keep reference to stream readers/writers per location. */
+  private final ConcurrentHashMap<BlockStoreLocation, Set<Object>> mStreamsPerLocation;
+  /** Used for delayed removing of streams in order to emulate activity cool-down. */
+  private final ScheduledExecutorService mScheduler;
```

Comment:
ManagementTaskCoordinator also have a cool down. However, it's required to implement the cool-down outside of coordinator as each task also need to track load by itself.

Reply:
Could you elaborate on this a bit more? Does the `LoadTracker` need to understand the load per task (ie. so we can do storage directory level throttling) or just whether or not there has been load?

Reply:
Could you elaborate on this a bit more? Does the LoadTracker need to understand the load per task (ie. so we can do storage directory level throttling) or just whether or not there has been load?

Reply:
`LoadTracker` needs to figure out if there was an external (user I/O) on a particular directory.
So it will be used by tasks to figure out if the directory that they're working on is under load, then back-off.